cdap-proto/src/main/java/co/cask/cdap/proto/id/NamespacedEntityId.java

```diff
@@ -26,6 +26,14 @@
 
   protected NamespacedEntityId(String namespace, EntityType entity) {
     super(entity);
+    if (namespace == null || namespace.isEmpty()) {
+      throw new IllegalArgumentException("Namespace can not be null or empty.");
+    }
+    if (!isValidNamespace(namespace)) {
+        throw new IllegalArgumentException(
+          String.format("Invalid characters found namespace ID %s. Allowed characters are letters, numbers, and _.",
```

Comment:
Sentence structure needs to be improved.
The first sentence doesn't sound right.